eth/core/state/statedb_test.go

```diff
@@ -116,4 +116,19 @@ var _ = Describe("StateDB", func() {
 		sdb.Finalize()
 		Expect(sdb.HasSuicided(bob)).To(BeFalse())
 	})
+
+	It("should have consistent gets and sets", func() {
+		key := common.Hash{0x01}
+		value := common.Hash{0x02}
+		addr := common.Address{}
+		value2 := common.Hash{0x03}
+
+		sdb.SetTransientState(addr, key, value)
+		Expect(sdb.GetTransientState(addr, key), value)
+		before := sdb.Snapshot()
+		sdb.SetTransientState(addr, key, value2)
+		Expect(sdb.GetTransientState(addr, key), value2)
+		sdb.RevertToSnapshot(before)
+		Expect(sdb.GetTransientState(addr, key), value)
+	})
```

Comment:
would recommend adding tests for the copy on the journal. Specifically, modifying the head entry on the journal should not modify the (head - 1)th entry on the journal

Reply:
also these tests can be in the journal package itself once this logic is removed from the statedb